nncf/quantization/algorithms/weight_compression/gptq.py

```diff
@@ -267,8 +269,14 @@ def _quantize_weights(
                         scales.append(scale)
                     else:
                         if self._scale_estimation and block_compression_config.num_bits == 4:
-                            activations = [inp[..., (i1 + i) : (i1 + i + group_size)] for inp in inputs]
-                            wc_statistics = ScaleEstimation.activations_to_wc_statistics(activations)
+                            activations = (
+                                [inp[..., (i1 + i) : (i1 + i + group_size), :] for inp in inputs]
+                                if input_channel_axis != (len(inputs[0].shape) - 1)
+                                else [inp[..., (i1 + i) : (i1 + i + group_size)] for inp in inputs]
+                            )
```

Comment:
Seems like if-condition is not needed. Can just slice along `input_channel_axis`:
```suggestion
                            slicing_along_axis = [slice(None)] * len(inputs[0].shape)
                            slicing_along_axis[input_channel_axis] = slice(i1 + i, i1 + i + group_size)
                            activations = [inp[tuple(slicing_along_axis)] for inp in inputs]
```

Reply:
Not sure why I was hesitant to use slice, this looks better :)